zerver/lib/actions.py

```diff
@@ -2010,6 +2011,20 @@ def do_change_left_side_userlist(user_profile, setting_value, log=True):
         log_event(event)
     send_event(event, [user_profile.id])
 
+def mark_user_star_backend(user_profile, email):
+    starred_user = get_object_or_404(UserProfile, email=email)
+    result = {}
+    if not starred_user in user_profile.starred_users.all():
+        user_profile.starred_users.add(starred_user)
+        result['starred'] = True
+    else:
+        user_profile.starred_users.remove(starred_user)
+        result['starred'] = False
+    user_profile.save()
+    result['starring_successful'] = True
+    result['email'] = email
```

Comment:
If you take advantage of closures on the javascript side, you shouldn't need to pass the email back. I just did a request--I know what email I specified in my request.

In the corresponding method in views/user_settings.py, you should just be able to return `json_success` unless you ran into a problem completing the request.